include/z64camera.h

```diff
@@ -0,0 +1,389 @@
+#ifndef Z64CAMERA_H
+#define Z64CAMERA_H
+
+#include "ultra64.h"
+
+// these two angle conversion macros are slightly inaccurate
+#define CAM_DEG_TO_BINANG(degrees) (s16)((degrees) * 182.04167f + .5f)
+#define CAM_BINANG_TO_DEG(binang) ((f32)(binang) * (360.0001525f / 65535.0f))
+
+#define NUM_CAMS 4
+
+// Camera IDs are indices into `cameraPtrs`
+#define CAM_ID_MAIN 0 // The index of the main camera
+#define CAM_ID_SUB_FIRST 1 // The index sub cameras start at
+#define CAM_ID_NONE -1 // Used to indicate no camera. Can be used to default to the active camera in some scenarios
+
+#define SUB_CAM_ID_DONE 0 // Used in some actors for variables holding sub camera IDs to indicate "subcam is finished"
+
+#define CAM_TRACKED_PLAYER(camera) ((Player*)camera->trackActor)
```

Comment:
I'm still not 100% sure on `trackActor` for this name. It's the actor that the camera is following, the "center of attention" if you will. In OoT, this is simply "player" but MM generalized this code to point to any actor (i.e. camera points at pamela or kafei as examples).

This also needs to be differentiated from `targetActor` which is a secondary actor that is being z-targeted.

What do you think about `followActor` perhaps? and maybe a macro `CAM_FOLLOW_PLAYER`? But then that macro sounds like it's applying something, maybe `GET_CAM_FOLLOW_PLAYER` idk. But yeah can add a comment elaborating on this

Reply:
`subject` is probably the technical term? Even `focusActor`?

Reply:
`subject` and `focus` are two terms already used in camera, `focusActor` makes me think of the "focus" of the actor, like `Actor_GetFocus` so that's weird. `subject` doesn't sound right (but that might just be me), and there are function types that are named after `Subject`, `Camera_SubjectX`, (technically `Camera_SubjX`), but that's based on original name and could be open to being changed. Problem is that those functions aren't well documented yet, so a good name change would have to wait until we know more about all the `Camera_SubjX` functions

Reply:
Hmmm, well maybe `focusActor` is worth considering. I'm thinking of a function like:

```
f32 Camera_GetTrackedActorHeight(Camera* camera) {
    PosRot actorFocus;
    Actor* trackActor = camera->trackActor;
    f32 trackHeight;

    if (trackActor == &GET_PLAYER(camera->play)->actor) {
        trackHeight = Player_GetHeight((Player*)trackActor);
    } else {
        Actor_GetFocus(&actorFocus, trackActor);
        trackHeight = actorFocus.pos.y - camera->trackActorPosRot.pos.y;
        if (trackHeight == 0.0f) {
            trackHeight = 10.0f;
        }
    }
    return trackHeight;
}
```

Maybe it could be something like:
```
f32 Camera_GetFocusActorHeight(Camera* camera) {
    PosRot actorFocus;
    Actor* focusActor = camera->focusActor;
    f32 focusHeight;

    if (focusActor == &GET_PLAYER(camera->play)->actor) {
        focusHeight = Player_GetHeight((Player*)focusActor);
    } else {
        Actor_GetFocus(&actorFocus, focusActor);
        focusHeight = actorFocus.pos.y - camera->focusActorPosRot.pos.y;
        if (focusHeight == 0.0f) {
            focusHeight = 10.0f;
        }
    }
    return trackHeight;
}
```
Is that too confusing?

Reply:
Maybe something like `CAMERA_TRACKED_AS_PLAYER`? The problem is it wasn't clear that it's "the tracked actor, cast to player"  if you see it elsewhere.

Reply:
I like `focusActor` or `focalActor`
Generally in 3d engines, cameras use the term focal point for what the camera is looking at.

Reply:
`focalActor` seems decent, to slightly differentiate it from `focus`. There's some docs I want to play around with that, so I'll take this line out so it doesn't hold up this PR, and bring it back in with the corresponding functions that it relates to